cpp/src/arrow/util/bit-util.cc

```diff
@@ -35,17 +35,19 @@
 
 namespace arrow {
 
-void BitUtil::FillBitsFromBytes(const std::vector<uint8_t>& bytes, uint8_t* bits) {
+namespace BitUtil {
+
+static void FillBitsFromBytes(const std::vector<uint8_t>& bytes, uint8_t* bits) {
```

Comment:
I've heard that using anonymous namespaces is preferable to `static`, because Clang/LLVM can do extra optimizations in some cases. I've been trying to stop using static except for static inline functions and static class members

Reply:
I'm a bit surprised, but I can do the change.
(note: I found https://stackoverflow.com/questions/154469/unnamed-anonymous-namespaces-vs-static-functions which doesn't mention a perf difference)